### DISCUSSION TITLE
Style/SignalException change my code from `fail` to `raise`, is there any meanful reason to do this?

### DISCUSSION THREAD

**Discussion by zw963** [rubocop/rubocop#10288] [Q&A]
```
Exceptions
raise vs fail

Prefer raise over fail for exceptions.

# bad
fail SomeException, 'message'

# good
raise SomeException, 'message'

```

What is the reason for this?

**Comment by koic** [comment]
JFYI, it seems to be based on the discussion in the Ruby Style Guide.

- https://github.com/rubocop/ruby-style-guide/issues/512
- https://github.com/rubocop/ruby-style-guide/pull/517
